### DISCUSSION TITLE
Use experimental rewrites cause page to reload

### DISCUSSION THREAD

**Discussion by DumbledoreD** [vercel/next.js#14512] [Help] [answered]
Trying out the experimental `rewrites` from https://github.com/vercel/next.js/discussions/9081. With the following config, both `<Link href="/abc">` and `router.push("/abc")` will make the page reload. `index` and `about` are SSR pages. Is this the expected behaviour? If so, is there a way to use `rewrites` but keep the app work like a SPA?

```js
// next.config.js
module.exports = {
  async rewrites() {
    return [
      { source: "/abc", destination: "/" },
      { source: "/bbc", destination: "/about" },
    ];
  }
};
```

**Comment by ijjk** [comment, marked as answer]
Hi, `rewrites` are currently only applied server-side not client-side so you still need to ensure you use matching `href` and `as` values e.g. `<Link href='/' as='/abc'><a>to /abc</a></Link>`

**Reply by DumbledoreD** [reply]
Thanks for the quick reply. Works like a charm.

**Comment by andresgutgon** [comment]
I'm having this same issue. Only that I'm using `rewrites` + `i18n`. 

The thing is that in development all looks that works. I see `.json` requests to the pages giving me fresh props. But when the app is in production mode I don't see the JSON requests and instead I get a full page reload. 

Any clues why this could happen?

In my case the route is called `houses.tsx`. but I want to be translated.

```
pages/houses.tsx

But users see:
/casas in Spanish (es)
```

This is my code:


``` javascript
// The path the user sees. 
router.push( 
  {  pathname: '/houses',   },      
  '/casas',      
  { scroll: false }    
)
```

**Reply by andresgutgon** [reply]
I had the rewrite with a `has` condition

[rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites):
```
{    
  source: '/casas',    
  has: [{ type: 'host', value: 'example.es' }],    
  destination: '/houses'  
}
```

So I removed `has` here and the page stopped doing full page reload.

**Comment by andresgutgon** [comment]
My question is: How can we pass `host` when doing `router.push`? Should we pass `host`? I'm talking in a i18n project with multiple hosts `.es`, `.it`, `.com`,... pointing to the same NextJS app. 

According to my tests removing the `has/post` in the `rewrite` fix the Full page reload problem.
